### ISSUE TITLE
Add Golang routes for reporting configuration differences to Traffic Ops in ort script

### ISSUE THREAD

**Issue by KevinMackenzie** [apache/incubator-trafficcontrol#1617]
It would be useful to have the ort script send the configuration differences between the cache's configuration files and those in the Traffic Ops database (in every mode except `REVALIDATE`).  This way the administrators of the CDN can see which/when caches are out of date and in what way.

@limited do you have anything to add?

**Comment by limited** [comment]
sounds good!

**Comment by mitchell852** [comment]
this feels like a snapshot diff but for caches. any thoughts on this @rob05c outside of simply running atstccfg locally on a cache to see the pending changes.

**Comment by rob05c** [comment]
Yeah, I think this would be a valuable feature. But it's a good amount of work. We'd need work on both the ORT side to get the diff and POST it to TO, and endpoint(s) on TO to receive that.

I'd +1 that. But that work should be coordinated with fixing the Snapshot-Queue issue a.k.a. "killing the chicken." Because fixing Snapshot-Queue also involves having ORT POST the date/snapshot that it applied back to TO. So, there might be some overlap, or maybe we could do both at the same time, or something.
